File: wf-beneficiary-management/template/java/model/request/InquiryBeneficiaryListRequest.java
/**
 * Alipay.com Inc.
 * Copyright (c) 2004-2026 All Rights Reserved.
 */
package {basePackage}.wf.model.request;

import org.apache.commons.lang3.builder.ToStringBuilder;
import org.apache.commons.lang3.builder.ToStringStyle;
import java.util.List;

/**
 * WorldFirst inquiryBeneficiaryList 请求对象。
 *
 * <p>用于分页查询已绑定的收款人列表。
 *
 * @author Qoder
 * @version InquiryBeneficiaryListRequest.java, v 0.1 2026-03-26
 */
public class InquiryBeneficiaryListRequest {

    /** 每页条数，最大 50 */
    private Integer pageSize;

    /** 页码（从 1 开始） */
    private Integer pageNumber;

    /** OAuth access token，最大 64 字符 */
    private String accessToken;

    /** 收款人令牌（精确匹配） */
    private String beneficiaryToken;

    /** 收款人昵称（模糊匹配） */
    private String beneficiaryNick;

    /** 集成商自定义唯一ID */
    private String referenceBeneficiaryId;

    /** 银行账号（模糊匹配） */
    private String bankAccountNo;

    /** IBAN */
    private String bankAccountIBAN;

    /** 币种过滤列表 */
    private List<String> currencyList;

    /** 账户名称（模糊匹配） */
    private String accountName;

    /** 资产类型：BANK_ACCOUNT 或 ALIPAY_ACCOUNT */
    private String assetType;

    /** 关系过滤：SAME_NAME 或 THIRD_PARTY */
    private List<String> relationFilter;

    public Integer getPageSize() {
        return pageSize;
    }

    public void setPageSize(Integer pageSize) {
        this.pageSize = pageSize;
    }

    public Integer getPageNumber() {
        return pageNumber;
    }

    public void setPageNumber(Integer pageNumber) {
        this.pageNumber = pageNumber;
    }

    public String getAccessToken() {
        return accessToken;
    }

    public void setAccessToken(String accessToken) {
        this.accessToken = accessToken;
    }

    public String getBeneficiaryToken() {
        return beneficiaryToken;
    }

    public void setBeneficiaryToken(String beneficiaryToken) {
        this.beneficiaryToken = beneficiaryToken;
    }

    public String getBeneficiaryNick() {
        return beneficiaryNick;
    }

    public void setBeneficiaryNick(String beneficiaryNick) {
        this.beneficiaryNick = beneficiaryNick;
    }

    public String getReferenceBeneficiaryId() {
        return referenceBeneficiaryId;
    }

    public void setReferenceBeneficiaryId(String referenceBeneficiaryId) {
        this.referenceBeneficiaryId = referenceBeneficiaryId;
    }

    public String getBankAccountNo() {
        return bankAccountNo;
    }

    public void setBankAccountNo(String bankAccountNo) {
        this.bankAccountNo = bankAccountNo;
    }

    public String getBankAccountIBAN() {
        return bankAccountIBAN;
    }

    public void setBankAccountIBAN(String bankAccountIBAN) {
        this.bankAccountIBAN = bankAccountIBAN;
    }

    public List<String> getCurrencyList() {
        return currencyList;
    }

    public void setCurrencyList(List<String> currencyList) {
        this.currencyList = currencyList;
    }

    public String getAccountName() {
        return accountName;
    }

    public void setAccountName(String accountName) {
        this.accountName = accountName;
    }

    public String getAssetType() {
        return assetType;
    }

    public void setAssetType(String assetType) {
        this.assetType = assetType;
    }

    public List<String> getRelationFilter() {
        return relationFilter;
    }

    public void setRelationFilter(List<String> relationFilter) {
        this.relationFilter = relationFilter;
    }

    @Override
    public String toString() {
        return ToStringBuilder.reflectionToString(this, ToStringStyle.SHORT_PREFIX_STYLE);
    }
    }
}

File: wf-api-integration/references/beneficiary/golang/model/request/inquiry_beneficiary_list_request.go
package request

import "fmt"

// InquiryBeneficiaryListRequest queries the list of bound beneficiaries with pagination.
type InquiryBeneficiaryListRequest struct {
	// PageSize is the number of items per page, max 50 (每页条数)
	PageSize int `json:"pageSize"`

	// PageNumber is the page number, starting from 1 (页码，从1开始)
	PageNumber int `json:"pageNumber"`

	// AccessToken is the OAuth access token, max 64 characters (OAuth访问令牌)
	AccessToken string `json:"accessToken,omitempty"`

	// BeneficiaryToken is for exact match filtering (收款人令牌，精确匹配)
	BeneficiaryToken string `json:"beneficiaryToken,omitempty"`

	// BeneficiaryNick is for fuzzy match filtering (收款人昵称，模糊匹配)
	BeneficiaryNick string `json:"beneficiaryNick,omitempty"`

	// ReferenceBeneficiaryID is the integrator's custom unique ID (集成商自定义唯一ID)
	ReferenceBeneficiaryID string `json:"referenceBeneficiaryId,omitempty"`

	// BankAccountNo is for fuzzy match filtering (银行账号，模糊匹配)
	BankAccountNo string `json:"bankAccountNo,omitempty"`

	// BankAccountIBAN is the IBAN for filtering (IBAN)
	BankAccountIBAN string `json:"bankAccountIBAN,omitempty"`

	// CurrencyList is the list of currencies for filtering (币种过滤列表)
	CurrencyList []string `json:"currencyList,omitempty"`

	// AccountName is for fuzzy match filtering (账户名称，模糊匹配)
	AccountName string `json:"accountName,omitempty"`

	// AssetType is the asset type: BANK_ACCOUNT or ALIPAY_ACCOUNT (资产类型)
	AssetType string `json:"assetType,omitempty"`

	// RelationFilter is the relationship filter: SAME_NAME or THIRD_PARTY (关系过滤)
	RelationFilter []string `json:"relationFilter,omitempty"`
}

// Validate validates the InquiryBeneficiaryListRequest.
func (r *InquiryBeneficiaryListRequest) Validate() error {
	if r.PageSize <= 0 || r.PageSize > 50 {
		return fmt.Errorf("pageSize must be between 1 and 50")
	}
	if r.PageNumber <= 0 {
		return fmt.Errorf("pageNumber must be >= 1")
	}
	if r.AccessToken != "" && len(r.AccessToken) > 64 {
		return fmt.Errorf("accessToken max length is 64")
	}
	return nil
}
